src/Workspaces/CSharp/Portable/Simplification/CSharpInferredMemberNameReducer.Rewriter.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.CSharp.Simplification
+{
+    internal partial class CSharpInferredMemberNameReducer
+    {
+        private class Rewriter : AbstractExpressionRewriter
+        {
+            public Rewriter(OptionSet optionSet, CancellationToken cancellationToken)
+                : base(optionSet, cancellationToken)
+            {
+            }
+
+            public override SyntaxNode VisitArgument(ArgumentSyntax node)
+            {
+                return SimplifyExpression(
+                    node,
+                    newNode: base.VisitArgument(node),
+                    simplifier: SimplifyTupleName);
```

Comment:
Unless something changed and we are now caching static methods, then this will be an allocation.  I was under the impression that we had not done the optimization to cache these yet.